src/pages/index.js

```diff
@@ -1,105 +1,284 @@
+        id: newUserData._id, //capture user id
+      });
+      profileEditForm.reset();
+    })
+  }
+  handleSubmit(makeRequest, profileEditFormPopup);
 }
 
-/* ------------------------------ Adding cards ------------------------------ */
+function handleAvatarEditSubmit(inputValue) {
+  const link = inputValue.link;
+  avatarEditSubmitButton.textContent = "Saving...";
+  function makeRequest() {
+    return api.updateProfileAvatar(link)
+    .then((newUserData) => {
+      avatarUserInfo.setAvatarInfo({ link: newUserData.avatar});
+      avatarEditForm.reset();
+    });
+  }
+  handleSubmit(makeRequest, avatarEditFormPopup);
 
-function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  return cardElement.getCardElement();
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  cardSection.addItem(cardElement);
+function handleNewItemSubmit(inputValues) {
+
+  function makeRequest() {
+    return api.addNewCard(inputValues.title, inputValues.link)
+    .then((newCardData) => {
+      renderCard(newCardData);
+      newItemPopup.close();
+      newItemForm.reset();
+    });
+  }
+  handleSubmit(makeRequest, newItemPopup);
+
 }
 
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
+function handleLikeCard(cardData) {
+  api
+    .likeCard(cardData._id)
+    .then(() => {
+      cardData.setCardLike(true);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+}
 
-cardSection.renderItems();
+function handleUnlikeCard(cardData) {
+  api
+    .unlikeCard(cardData._id)
+    .then(() => {
+      cardData.setCardLike(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+}
 
-const newItemPopup = new PopupWithForm({
-  popupSelector: "#new-item-modal",
-  handleFormSubmit: handleNewItemSubmit,
+function handleDeleteModal(cardData) {
+  formValidators["delete-form"].resetValidation();
+  deleteCardPopup.open();
+  deleteCardPopup.setFormSubmitHandler(() => {
+    api
+      .deleteCard(cardData._id) //replaced cardId with cardData._id
+      .then(() => {
+        cardData.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch(console.error);
+  });
+
+}
+
+function handleImageClick(cardData) {
+  previewImagePopup.open(cardData);
+}
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Listeners                              */
+/* -------------------------------------------------------------------------- */
+
+profileEditFormPopup.setEventListeners();
+
+profileEditButton.addEventListener("click", function () {
+  const { name, description } = profileUserInfo.getUserInfo();
+  profileTitleInput.value = name;
+  profileDescriptionInput.value = description;
+  formValidators["profile-edit-form"].resetValidation();
+  formValidators["profile-edit-form"].disableButton();
```

Comment:
`formValidators["profile-edit-form"].disableButton();` is not needed here


To disable the button you need to place `resetValidation` before you set the inputs data

```js
profileEditButton.addEventListener("click", function () {
  formValidators["profile-edit-form"].resetValidation();  // here
  const { name, description } = profileUserInfo.getUserInfo();
  profileTitleInput.value = name;
  profileDescriptionInput.value = description;
  
  profileEditFormPopup.open();
});

```